docs/getting-started/built-ins.md

```diff
@@ -153,57 +153,153 @@ rshift(1024n, 11n) // 0n
 
 - `exit(status: boolean): void` Call this function will terminate contract execution. If `status` is `true` then the contract succeeds; otherwise, it fails.
 
+## `SmartContract` Methods
+
+The following `@methods` come with the `SmartContract` base class.
+
+### `checkSig`
+
+Function `checkSig(signature: Sig, publicKey: PubKey): boolean` verifies an ECDSA signature. It takes two inputs: an ECDSA signature and a public key. 
+
+It returns true if the signature matches the public key. Returns false if the signature is an empty. Otherwise, the entire contract fails immediately, due to the [**NULLFAIL** rule](https://github.com/bitcoin/bips/blob/master/bip-0146.mediawiki#NULLFAIL).
+
+For example, Pay-to-Public-Key-Hash ([P2PKH](https://learnmeabitcoin.com/guide/p2pkh)) can be implemented as below.
+
+```ts
+class P2PKH extends SmartContract {
+  // public key hash of the recipient.
+  @prop()
+  readonly pubKeyHash: PubKeyHash
+
+  constructor(pubKeyHash: PubKeyHash) {
+    super(...arguments)
+    this.pubKeyHash = pubKeyHash
+  }
+
+  @method()
+  public unlock(sig: Sig, pubkey: PubKey) {
+    // check if the passed public key belongs to the specified public key hash
+    assert(hash160(pubkey) == this.pubKeyHash, 'public key hashes are not equal')
+    // check signature validity
+    assert(this.checkSig(sig, pubkey), 'signature check failed')
+  }
+}
+```
+
+### `buildStateOutput`
+
+Function `buildStateOutput(amount: bigint): ByteString` creates an output containing the latest state. It takes an input: the number of satoshis in the output.
+
+```ts
+class Counter extends SmartContract {
+  // ...
+
+  @method(SigHash.ANYONECANPAY_SINGLE)
+  public incOnChain() {
+    // ... update state
+      
+    // construct the new state output 
+    const output: ByteString = this.buildStateOutput(this.ctx.utxo.value)
+
+    // ... verify outputs of current tx
+  }
+}
+```
+
+### `buildChangeOutput`
+
+Function `buildChangeOutput(): ByteString` creates a P2PKH change output. It will calculate the change amount (`this.changeAmount`) automatically, and use the signer's address by default, unless `changeAddress` field is explicitly set in `MethodCallOptions`.
```

Comment:
if the contract `@method` doesn't not have a transation builder,  the change address can be set by the `changeAddress` field in `MethodCallOptions`. Otherwise, it can be set by calling  `tx.change()` for a `Transaction`